data/ai/lua/ca_village_hunt.lua

```diff
@@ -60,16 +60,16 @@ function ca_village_hunt:evaluation(cfg, data, filter_own)
         return 0
     end
 
-    local avoid_map = AH.get_avoid_map(ai, nil, true)
+    local avoid_map2 = AH.get_avoid_map(ai, nil, true)
```

Comment:
Without knowing anything about what these changes are doing, is there something more descriptive we can use than just appending numbers? Is it just to make the variables unique and for some reason we cannot re-use existing variables?

Reply:
The warning being fixed is about a variable shadowing another, which isn't good both for readability as well as if you remove the second declaration of the variable then all the code that was previously using it will now silently start using the firs declaration rather than warning about using an undefined variable.

Reply:
Hmm, okay. Generally I try to avoid calling things `x1`, `x2`, `x3`, etc, but I suppose it'd be a bit much to try to give everything sensible names here.

Reply:
I think it would be better to have the config for luacheck, but for the moment have it as something that developers (backed up by code-reviewers) run manually, with a long-term goal of looking carefully at all of the warnings.

With the code that we're commenting on, I think both `avoid_map1` and `avoid_map2` will have the same contents:
```
    local avoid_map1 = LS.of_pairs(ai.aspects.avoid)
    local avoid_map2 = AH.get_avoid_map(ai, nil, true)
```
but that `get_avoid_map(ai, nil, true)` appears to also return `LS.of_pairs(ai.aspects.avoid)`.

Reply:
I don't see how those are related? The goal here isn't to evaluate every single one of hundreds of warnings to determine whether the current code isn't ideal in some way, it's to address the existing warnings so that the check can be run in the CI to prevent even more of the same types of problems from being added.

Reply:
You're right - I was imagining a dream-world where everyone rigorously opened PRs for all changes.